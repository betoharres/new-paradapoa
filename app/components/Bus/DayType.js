import React, {useState} from 'react'
import {FlatList, Dimensions} from 'react-native'
import {shape, object} from 'prop-types'
import Carousel from 'react-native-snap-carousel'

import {Container, Card, ListItem} from './DayType.styles'
import Pagination from './Pagination'

import {dayTypeLabels} from './lib/constants'
import {getCurrentDayType} from './lib/utils'

const {width: SCREEN_WIDTH} = Dimensions.get('window')

export default function DayType({
  route: {
    params: {dayTypeSchedules},
  },
}) {
  const [activeDayType, setActiveDayType] = useState(0)
  const schedules = Object.keys(dayTypeSchedules)
  const currentDayType = getCurrentDayType(schedules)
  return (
    <>
      <Container>
        <Carousel
          data={schedules}
          firstItem={currentDayType}
          sliderWidth={SCREEN_WIDTH}
          swipeThreshold={5}
          itemWidth={SCREEN_WIDTH * 0.55}
          inactiveSlideOpacity={0.85}
          inactiveSlideScale={0.6}
          onSnapToItem={setActiveDayType}
          renderItem={({item: dayType, index}) => {
            return (
              <Card key={dayType} title={dayTypeLabels[dayType]}>
                <FlatList
                  data={dayTypeSchedules[dayType]}
                  keyExtractor={({id}) => String(id)}
                  renderItem={({item: {time}}) => <ListItem title={time} />}
                />
              </Card>
            )
          }}
        />
      </Container>
      <Pagination entries={schedules} activeIndex={activeDayType} />
    </>
  )
}

DayType.propTypes = {
  route: shape({
    params: shape({
      dayTypeSchedules: object.isRequired,
    }).isRequired,
  }).isRequired,
}
