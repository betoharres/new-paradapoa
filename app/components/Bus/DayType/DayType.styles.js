import {Dimensions} from 'react-native'
import styled from 'styled-components/native'
import {Card as RNECard, ListItem as RNEListItem} from 'react-native-elements'

const {height: SCREEN_HEIGHT} = Dimensions.get('window')

export const Container = styled.View`
  flex: 1;
  flex-direction: column;
  align-items: center;
`

export const Card = styled(RNECard).attrs(() => ({
  containerStyle: {
    borderRadius: 15,
    maxHeight: SCREEN_HEIGHT - 205,
    paddingBottom: 50,
  },
  dividerStyle: {
    display: 'none',
  },
}))``

export const ListItem = styled(RNEListItem).attrs(({disabled}) => ({
  titleStyle: {
    color: disabled ? 'grey' : 'black',
  },
}))``
