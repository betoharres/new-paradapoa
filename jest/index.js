export function mockProps({props = {}, params = {}} = {}) {
  const {navigation} = props;
  const navigate = (navigation && navigation.navigate) || jest.fn();
  const goBack = (navigation && navigation.goBack) || jest.fn();
  const setParams = (navigation && navigation.setParams) || jest.fn();
  const mockNavigation = {navigation: {navigate, goBack, setParams}};
  return {...{route: {params}}, ...props, ...mockNavigation};
}

export * from '@testing-library/react-native';
