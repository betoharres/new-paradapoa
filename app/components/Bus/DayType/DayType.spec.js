import React from 'react';
import DayType from './DayType';
import {render, mockProps} from '~/jest';
// import schedules from '~/jest/665.json';

it('renders correctly', () => {
  const props = mockProps({params: {dayTypeSchedules: {2: {2: []}}}});
  const wrapper = render(<DayType {...props} />);
  expect(wrapper.toJSON()).toMatchSnapshot();
});
