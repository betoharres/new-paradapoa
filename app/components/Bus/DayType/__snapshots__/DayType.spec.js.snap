// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    activeAnimationOptions={null}
    activeAnimationType="timing"
    activeSlideAlignment="center"
    activeSlideOffset={20}
    apparitionDelay={0}
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDelay={1000}
    autoplayInterval={3000}
    callbackOffsetMargin={5}
    containerCustomStyle={Object {}}
    contentContainerCustomStyle={Object {}}
    contentContainerStyle={
      Array [
        Object {
          "paddingLeft": 180,
          "paddingRight": 180,
        },
        Object {},
      ]
    }
    data={
      Array [
        "2",
      ]
    }
    decelerationRate="fast"
    directionalLockEnabled={true}
    disableVirtualization={false}
    enableMomentum={false}
    enableSnap={true}
    firstItem={0}
    forwardedRef={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    hasParallaxImages={false}
    horizontal={true}
    inactiveSlideOpacity={0.85}
    inactiveSlideScale={0.6}
    inactiveSlideShift={0}
    initialNumToRender={7}
    invertStickyHeaders={false}
    inverted={false}
    itemWidth={390}
    keyExtractor={[Function]}
    layout="default"
    lockScrollTimeoutDuration={1000}
    lockScrollWhileSnapping={false}
    loop={false}
    loopClonesPerSide={3}
    maxToRenderPerBatch={15}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onResponderRelease={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onSnapToItem={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    overScrollMode="never"
    pinchGestureEnabled={false}
    removeClippedSubviews={true}
    renderItem={[Function]}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    shouldOptimizeUpdates={true}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    slideStyle={Object {}}
    sliderWidth={750}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "flexDirection": "row",
        "opacity": 0,
        "width": 750,
      }
    }
    swipeThreshold={5}
    updateCellsBatchingPeriod={50}
    useScrollView={false}
    vertical={false}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={15}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "opacity": 1,
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
              "width": 390,
            }
          }
        >
          <View
            dayType="2"
            style={
              Object {
                "backgroundColor": "white",
                "borderColor": "#e1e8ee",
                "borderRadius": 15,
                "borderWidth": 1,
                "elevation": 1,
                "margin": 15,
                "marginBottom": 0,
                "maxHeight": 1129,
                "padding": 15,
                "paddingBottom": 50,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "transparent",
                }
              }
            >
              <View>
                <Text
                  style={
                    Object {
                      "color": "#43484d",
                      "fontFamily": "sans-serif",
                      "fontSize": 17.5,
                      "fontWeight": "bold",
                      "marginBottom": 15,
                      "textAlign": "center",
                    }
                  }
                  testID="cardTitle"
                  theme={
                    Object {
                      "colors": Object {
                        "disabled": "hsl(208, 8%, 90%)",
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "platform": Object {
                          "android": Object {
                            "error": "#f44336",
                            "primary": "#2196f3",
                            "secondary": "#9C27B0",
                            "success": "#4caf50",
                            "warning": "#ffeb3b",
                          },
                          "ios": Object {
                            "error": "#ff3b30",
                            "primary": "#007aff",
                            "secondary": "#5856d6",
                            "success": "#4cd964",
                            "warning": "#ffcc00",
                          },
                        },
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    }
                  }
                >
                  SÃ¡bado
                </Text>
                <View
                  style={
                    Object {
                      "backgroundColor": "#99BBE5",
                      "height": 1,
                      "marginBottom": 15,
                    }
                  }
                />
              </View>
              <RCTScrollView
                data={
                  Object {
                    "2": Array [],
                  }
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={true}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View />
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
